### PR TITLE
Improve accessibility on the search module

### PR DESCRIPTION
**Pull request by mdaif**
Frontend is not really my thing but I had to do something about a 2 year old bug :)) 
This pull request aims to:
1. adds label for search-q
2. adds text to search button
3. adds role div 

See https://bugzilla.mozilla.org/show_bug.cgi?id=764345

### CODE REVIEW

File: templates/impala/search.html [hunk 1]
```diff
@@ -16,7 +19,10 @@
     {{ search_form.platform }}
   {% endif %}
   <button class="search-button" type="submit" title="{{ _('Search') }}"
-         src="{{ static('img/zamboni/global/btn-search.png') }}"></button>
+         src="{{ static('img/zamboni/global/btn-search.png') }}"
+          style="text-indent:-999em;">search
```
Comment (muffinresearch) [inline templates/impala/search.html]: We'd prefer to not add inline CSS. There might be a util class already in our CSS that does this or similar.

File: templates/impala/search.html [hunk 2]
```diff
@@ -3,7 +3,10 @@
   {% set search_placeholder = 'all' %}
 {% endif %}
 {% set search_form = SimpleSearchForm(request, search_cat) %}
+
 <form id="search" action="{{ search_url }}">
+    <div role="status" aria-live="polite" id="announce-suggestions"></div>
+    <label for="search-q" hidden>type to search for add-ons</label>
```
Comment (muffinresearch) [inline templates/impala/search.html]: The hidden attribute means this won't be visible to assistive tech as well as user agents. In addition the label text needs to be setup to be localized using `{{ _('text here) }}`.

Comment (mdaif) [inline templates/impala/search.html]: I'll work on it, but may I ask what is the best way to simulate a screen reader ?

File: templates/impala/search.html (hunk 1 above)
Comment (mdaif) [inline templates/impala/search.html]: I am completely lost in css files, most of them are read-only and I can't find a class that hides text, so I need to create a new one, which css file should I define the new class at ?

File: templates/impala/search.html (hunk 2 above)
Comment (muffinresearch) [inline templates/impala/search.html]: Running a real one is likely going to be a better approach than trying to use a simulator if you want to try it out. Here's a couple of articles written from the perspective of testing (both for windows) with JAWs http://webaim.org/articles/jaws/ and another for using NVDA: http://webaim.org/articles/nvda/ . If you're on a mac then you can try with voiceover. http://webaim.org/articles/voiceover/

File: templates/impala/search.html (hunk 1 above)
Comment (muffinresearch) [inline templates/impala/search.html]: > I am completely lost in css files, most of them are read-only and I can't find a class that hides text, so I need to create a new one, which css file should I define the new class at ?

So I would add something like this [1] (from html5bp)  from a quick glance I don't see anything similar in our code base. This is a 'current' approach for hiding text but allowing it to be read by screen readers - you'll need to wrap the text in a span and apply a class to that for this approach though.

It looks like this file [2] is the best place for site-wide styling 

[1] https://github.com/h5bp/html5-boilerplate/blob/0c8b71eb88d9bcc8656fe3cecc76ccc3cd30c310/dist/css/main.css#L131
[2] https://github.com/mozilla/olympia/blob/master/static/css/impala/site.less

**Comment by mdaif** [comment]
Can you check the latest modifications ?

**Comment by muffinresearch** [comment]
@mdaif sorry for the delay - holidays got in the way. Looking now.

File: static/js/impala/ajaxcache.js [hunk 3]
```diff
@@ -92,6 +92,9 @@ $.ajaxCache = function(o) {
             }
         }
     }
+    $("#live-announcements").html(gettext(
```
Comment (muffinresearch) [inline static/js/impala/ajaxcache.js]: This doesn't need to use `html()` does it? Could just be `text()`?

File: static/js/impala/ajaxcache.js [hunk 4]
```diff
@@ -92,6 +92,9 @@ $.ajaxCache = function(o) {
...
+    "One or more suggestion is available You can select between them"));
```
Comment (muffinresearch) [inline static/js/impala/ajaxcache.js]:
```udiff
- "One or more suggestion is available You can select between them"));
+ "One or more suggestions is available. You can select between them."));
```

File: templates/impala/search.html [hunk 5]
```diff
@@ -16,7 +22,11 @@
     {{ search_form.platform }}
   {% endif %}
   <button class="search-button" type="submit" title="{{ _('Search') }}"
-         src="{{ static('img/zamboni/global/btn-search.png') }}"></button>
+         src="{{ static('img/zamboni/global/btn-search.png') }}">
+      <span class="visuallyhidden">
+          {{_('search for add-ons')}}
```
Comment (muffinresearch) [inline templates/impala/search.html]: This could probably simply say 'Search'.

File: static/js/impala/ajaxcache.js (hunk 3 above)
Comment (muffinresearch) [inline static/js/impala/ajaxcache.js]: How does this string get cleared?

File: locale/en_US/LC_MESSAGES/javascript.po [hunk 6]
```diff
@@ -800,6 +800,7 @@ msgstr "Date"
 msgid "Other"
 msgstr "Other"
 
+
```
Comment (muffinresearch) [inline locale/en_US/LC_MESSAGES/javascript.po]: Unecessary additional line.

File: static/js/impala/ajaxcache.js (hunk 3 above)
Comment (muffinresearch) [inline static/js/impala/ajaxcache.js]: Our coding style generally prefers using single quotes over double-quotes in JS.

Comment (mdaif) [inline static/js/impala/ajaxcache.js]: The example yahoo page doesn't clear the string ... When do we need to clear the string ? a use case I mean

**Comment by muffinresearch** [comment]
> The example yahoo page doesn't clear the string ... When do we need to clear the string ? a use case I mean

When search results are returned the text is updated which should be read out due to population of the element with `role="status"`. If this is filled in does it need to be cleared when a new search takes place or if the text field is blurred (when the find as you type search results are hidden)? 

In other words this just fills in a status once and doesn't continue to reflect the status of the search.

**Comment by muffinresearch** [comment]
@mdaif this needs rebasing against master to fix conflicts.

Any thoughts on https://github.com/mozilla/olympia/pull/1074#issuecomment-168978755

**Comment by muffinresearch** [comment]
Going to close this as it's bit-rotted. Please reopen once the conflicts are resolved we'll be happy to continue the review process.

**Pull request closed without merging**